proposals/p4105.md

```diff
@@ -0,0 +1,280 @@
+be incrementally tackled.
+
+### Long-Term Stable (LTS) versions and standardization
+
+SemVer alone isn't sufficient to address some user needs for language stability.
+It is enough to _detect_ the issues when they arise, but Carbon should also plan
+for how to _address_ these issues.
+
+The suggested direction here is towards designated LTS versions based on a
+particular level of completeness and quality and the user demand. These versions
+will likely need even longer time horizons of support than Linux distro LTS
+releases. The direction should be to embrace this and the potential for
+multi-decade support windows to support users' needs. As the windows of LTSes
+expand, their frequency should reduce to avoid supporting an unsustainable
+diversity of versions.
+
+Exactly how a version is designated as LTS is left to the future work here, but
+it should not be expected to change the schema and structure of the versioning,
+just the support policy applied to the specific release version in question.
+
+Some users may even require standardization of a programming language to make it
+usable in their environment. Carbon should again embrace this need and see the
+standardization as an analogous process to promoting a normal release into an
+LTS. Some relevant and effective LTS should be selected and taken through
+whatever process is identified to create a standard reflecting that LTS version
+of Carbon. Updates to the standard should in turn track as updates to a newer
+LTS. The specifics of how to do this are left to the future work, and they may
+change exactly how this works.
+
+Note that the goal of this future direction isn't to constrain how Carbon can
+arrive effectively at either an LTS release or a standard. Instead, the goal is
+to make it clear that we _should_ be open and planning to achieve these in order
+to meet the needs of candidate Carbon users.
+
+## Rationale
+
+-   [Language tools and ecosystem](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md#language-tools-and-ecosystem)
+    -   Carbon needs a coherent versioning scheme across the language itself as
+        well as its ecosystem. Especially as the language is developing rapidly,
+        being able to sync across all of these with a single, simple versioning
+        scheme is especially important to have the tooling and ecosystem agree
+        about features of the language.
+-   [Software and language evolution](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md#software-and-language-evolution)
+    -   Users of Carbon need a clear model for understanding the language's
+        evolution, tracking it, and responding to significant aspects of it.
+    -   The versioning scheme needs to support a wide variety of versions built
+        in the process of evolving the language without creating confusion.
+
+## Alternatives considered
```

Comment:
Is it worth mentioning the alpha/beta discussion? I think they were in the docs version of the proposal, but removed here. Wondering if they should be mentioned for posterity.

Reply:
Good call, added.